src/tests/pools/D3MAaveV3Pool.t.sol

```diff
@@ -0,0 +1,375 @@
+    }
+
+    function setScaledTotalSupply(uint256 amount) external {
+        scaledTotalSupply = amount;
+    }
+
+    function getIncentivesController() external view returns (address) {
+        return rewardsClaimer;
+    }
+}
+
+contract FakeRewardsClaimer {
+    struct ClaimCall {
+        address[] assets;
+        uint256 amt;
+        address dst;
+        address reward;
+    }
+    ClaimCall public lastClaim;
+
+    function claimRewards(address[] calldata assets, uint256 amt, address dst, address reward) external returns (uint256) {
+        lastClaim = ClaimCall(
+            assets,
+            amt,
+            dst,
+            reward
+        );
+        return amt;
+    }
+
+    function getAssetsFromClaim() external view returns (address[] memory) {
+        return lastClaim.assets;
+    }
+}
+
+contract FakeLendingPool {
+
+    // Need to use a struct as too many variables to return on the stack
+    struct ReserveData {
+        //stores the reserve configuration
+        uint256 configuration;
+        //the liquidity index. Expressed in ray
+        uint128 liquidityIndex;
+        //the current supply rate. Expressed in ray
+        uint128 currentLiquidityRate;
+        //variable borrow index. Expressed in ray
+        uint128 variableBorrowIndex;
+        //the current variable borrow rate. Expressed in ray
+        uint128 currentVariableBorrowRate;
+        //the current stable borrow rate. Expressed in ray
+        uint128 currentStableBorrowRate;
+        //timestamp of last update
+        uint40 lastUpdateTimestamp;
+        //the id of the reserve. Represents the position in the list of the active reserves
+        uint16 id;
+        //aToken address
+        address aTokenAddress;
+        //stableDebtToken address
+        address stableDebtTokenAddress;
+        //variableDebtToken address
+        address variableDebtTokenAddress;
+        //address of the interest rate strategy
+        address interestRateStrategyAddress;
+        //the current treasury balance, scaled
+        uint128 accruedToTreasury;
+        //the outstanding unbacked aTokens minted through the bridging feature
+        uint128 unbacked;
+        //the outstanding debt borrowed against this asset in isolation mode
+        uint128 isolationModeTotalDebt;
+    }
+
+    address public adai;
+    uint256 public supplyCap;
+    uint256 public liquidityIndex = 10 ** 27;
+    uint256 public accruedToTreasury;
+
+    struct DepositCall {
+        address asset;
+        uint256 amt;
+        address forWhom;
+        uint16 code;
+    }
+    DepositCall public lastDeposit;
+
+    struct WithdrawCall {
+        address asset;
+        uint256 amt;
+        address dst;
+    }
+    WithdrawCall public lastWithdraw;
+
+    constructor(address adai_) {
+        adai = adai_;
+    }
+
+    function getReserveData(address) external view returns(
+        ReserveData memory result
+    ) {
+        result.configuration = supplyCap << 116;
```

Comment:
No issue here, just wondering why we didn't do the 1 2 3 4 5 6 method that v2 did? Especially with the << 116 part when we could have done '0' like v2 (unless we couldn't have done that?). I'm a bit lost here so this comment is just flagging it for checking by someone who is more familiar with this section.

Reply:
Not sure I understand what you are getting at. This is setting up the configuration which has supply cap starting at bit 116.

Reply:
I took a look at it again and it looks OK to me now. This can be resolved.